Add template for release PR (#509)
precice/precice: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md

## Step by step guide

> * [ ] Merge master to develop (No commits after the release on master) @MakisH 

Reviewer: In the template we only want the steps. The people change all the time.

* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`. @MakisH 
* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)) @MakisH 

> * [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) and add things if necessary and extract summary (all) @fsimonis 

Author: Added this in 
f1e50e1

* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)

> * [ ] Bump version in CMakeLists.txt, SConstruct, and CHANGELOG (and Python bindings). @MakisH 

Author: Added this in 
5d0535b

* [ ] Draft message to mailing list @uekerman @MakisH 
* [ ] Update documentation (all) @MakisH 

> * [ ] Fix potential problems in develop (all)

Reviewer: ```suggestion
* [ ] Fix potential problems in develop (all)
* [ ] Approve the PR with at least two reviews (all)
```

> * [ ] Merge develop to master @MakisH 

Reviewer: ```suggestion
* [ ] Merge PR to master @MakisH 
```

> * [ ] Tag release (on GitHub) and merge back to develop @MakisH 

Reviewer: ```suggestion
* [ ] Tag release (on GitHub, make sure to select the release branch as target) and merge back to develop
```

## Regression Tests

> Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289

Reviewer: That's a very good note! Apart from the revisions, I would also add more information, as e.g. in [this example](https://github.com/precice/precice/pull/507#issuecomment-530409554).

Reviewer: I think that linking to an example is enough to get the idea. Let's not overengineer this.

* [ ] SU2/CalculiX @MakisH 
* [ ] OpenFOAM / OpenFOAM @MakisH 
* [ ] OpenFOAM / OpenFOAM - NP mapping in OpenFOAM @MakisH
* [ ] OpenFOAM / OpenFOAM fluid coupling module @MakisH 
* [ ] OpenFOAM / CalculiX FSI (flap perp) @MakisH
* [ ] OpenFOAM / CalculiX FSI - NP mapping in CalculiX (3DTube) @KyleDavisSA 
* [ ] OpenFOAM / CalculiX / OpenFOAM CHT (heat exchanger) @MakisH
* [ ] OpenFOAM / deal.II @MakisH 

> * [ ] OpenFOAM / FEniCS [flap_perp](https://github.com/precice/tutorials/tree/master/FSI/flap_perp/OpenFOAM-FEniCS) @BenjaminRueth 

Reviewer: Links are great! Let's add links also in the other cases.

Author: I think some links are still missing here.

## How to work with this template

* [ ] assign a release manager, who takes care of the process

> * [ ] assign each point below to a responsible person, before you continue. Use `@member`.

Reviewer: ```suggestion
* [ ] assign each point below to a responsible person, before you continue. Use `@member`.

Only the release manager should update this post (even tickboxes, due to race conditions in editing). Everybody else should comment on the PR with the progress.
```

## Step by step guide
* [ ] Merge master to develop (No commits after the release on master)
* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`.
* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md))

> * [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) and add things if necessary and extract summary (all)

Reviewer: ```suggestion
* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) (all)
   * Add things, if necessary
   * Extract summary
```

* [ ] Look over the Roadmap and update entries.
* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
* [ ] Bump version in:
   * [ ] [CHANGELOG](https://github.com/precice/precice/blob/develop/CHANGELOG.md)
   * [ ] [CMakeLists.txt](https://github.com/precice/precice/blob/develop/CMakeLists.txt)

>    * [ ] **???** [Python bindings](https://github.com/precice/python-bindings) 

Reviewer: Still open.

* [ ] SU2/CalculiX
* [ ] OpenFOAM / OpenFOAM
* [ ] OpenFOAM / OpenFOAM - NP mapping in OpenFOAM
* [ ] OpenFOAM / OpenFOAM fluid coupling module 
* [ ] OpenFOAM / CalculiX FSI (flap perp)
* [ ] OpenFOAM / CalculiX FSI - NP mapping in CalculiX (3DTube)
* [ ] OpenFOAM / CalculiX / OpenFOAM CHT (heat exchanger)
* [ ] OpenFOAM / deal.II
* [ ] OpenFOAM / FEniCS [flap_perp](https://github.com/precice/tutorials/tree/master/FSI/flap_perp/OpenFOAM-FEniCS)
* [ ] OpenFOAM / FEniCS [flow-over-plate](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-fenics)
* [ ] OpenFOAM / Nutils [flow-over-plate](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-nutils)
* [ ] FEniCS / FEniCS [partitioned-heat](https://github.com/precice/tutorials/tree/master/HT/partitioned-heat/fenics-fenics)
* [ ] ExaFSA: Ateles / FASTEST
* [ ] Alya
* [ ] 1D-ElasticTube (C++)
* [ ] 1D-ElasticTube (Python)
* [ ] SuperMUC
* [ ] C++ Dummy
* [ ] C Dummy
* [ ] Fortran Dummy
* [ ] Fortran 2003 Dummy
* [ ] Python Dummy

## Post-release
* [ ] Generate packages
   * [ ] Ubuntu 18.04

>    * [ ] Ubuntu 19.04

Reviewer: ```suggestion
   * [ ] Ubuntu 19.10
```

   * [ ] Arch Linux AUR Package
* [ ] Update Spack recipe
* [ ] Send email and do marketing

>    * [ ] Mention [xSDK](https://github.com/xsdk-project/xsdk-policy-compatibility/blob/master/precice-policy-compatibility.md)

Reviewer: ```suggestion
```
(This was relevant only in v1.6.0)

* [ ] Tweet
* [ ] Format the code base

> * [ ] Update the [PR template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)?

Reviewer: My main question is how do we make sure that we don't always forget how to create a new PR with this template. I suggest adding the following:

```suggestion
* [ ] Update the [PR template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)

To open a new PR with this template, use this [PR template query](https://github.com/precice/precice/compare/new?template=release_pull_request_template.md)
```

Only the release manager should update this post (even tickboxes, due to race conditions in editing). Everybody else should comment on the PR with the progress.

## Step by step guide
* [ ] Merge master to develop (No commits after the release on master)
* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`.
* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md))
* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) (all)
   * Add things, if necessary
   * Extract summary
* [ ] Look over the Roadmap and update entries.
* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
* [ ] Bump version in:
   * [ ] [CHANGELOG](https://github.com/precice/precice/blob/develop/CHANGELOG.md)
   * [ ] [CMakeLists.txt](https://github.com/precice/precice/blob/develop/CMakeLists.txt)
   * [ ] **???** [Python bindings](https://github.com/precice/python-bindings) 
* [ ] Draft message to mailing list

> * [ ] Update documentation (all)

Reviewer 2: ```suggestion
* [ ] Update documentation (all)
  * [ ] Update markdown configuration reference in wiki
```

Reviewer 2: see #571

   * [ ] (do independent release, if necessary!) [MATLAB bindings](https://github.com/precice/matlab-bindings)
* [ ] Draft message to mailing list
* [ ] Update documentation (all)
  * [ ] Update markdown configuration reference in wiki
* [ ] Fix potential problems in develop (all)
* [ ] Approve the PR with at least two reviews (all)
* [ ] Merge PR to master 
* [ ] Tag release (on GitHub, make sure to select the release branch as target) and merge back to develop

## Regression Tests

Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289

* [ ] SU2 / CalculiX [flap_perp](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp/SU2-CalculiX)
* [ ] OpenFOAM / OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)
* [ ] OpenFOAM / OpenFOAM - NP mapping in OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)

> * [ ] OpenFOAM / OpenFOAM fluid coupling module **???**

Reviewer: This is not really ready for regular testing, it is one of the many under-development cases we have. Let's leave it out for now.
```suggestion
```

* [ ] OpenFOAM / Nutils [flow-over-plate](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-nutils)
* [ ] FEniCS / FEniCS [partitioned-heat](https://github.com/precice/tutorials/tree/master/HT/partitioned-heat/fenics-fenics)
* [ ] ExaFSA: Ateles / FASTEST
* [ ] Alya
* [ ] 1D-ElasticTube [C++](https://github.com/precice/elastictube1d/tree/develop/cxx)
* [ ] 1D-ElasticTube [Python](https://github.com/precice/elastictube1d/tree/develop/python)
* [ ] SuperMUC
* [ ] Solverdummy [C++](https://github.com/precice/precice/tree/develop/tools/solverdummies/cpp)
* [ ] Solverdummy [C](https://github.com/precice/precice/tree/develop/tools/solverdummies/c)
* [ ] Solverdummy [Fortran](https://github.com/precice/precice/tree/develop/tools/solverdummies/fortran)
* [ ] Solverdummy [Fortran 2003](https://github.com/precice/precice/tree/develop/tools/solverdummies/f2003)
* [ ] Solverdummy [Python](https://github.com/precice/python-bindings/tree/develop/solverdummy)
* [ ] Solverdummy [MATLAB](https://github.com/precice/matlab-bindings/tree/develop/solverdummy)

## Post-release
* [ ] Generate packages

>    * [ ] Ubuntu 18.04

Reviewer: ```suggestion
   * [ ] Latest Ubuntu LTS
```

>    * [ ] Ubuntu 19.10

Reviewer: ```suggestion
   * [ ] Latest Ubuntu
```

   * [ ] Arch Linux AUR Package
* [ ] Update Spack recipe
* [ ] Send email and do marketing
* [ ] Tweet

> * [ ] Format the code base

Reviewer: If we want to keep this in a regular basis, we should move it in the beginning, before testing.

Author: Fixed https://github.com/precice/precice/pull/509/commits/b3cba80002021c939e09faebe306835cae45c4da
